tests/algorithms/test_colout.py

```diff
@@ -107,134 +106,130 @@ def fake_image_batch(H: int, W: int, C: int) -> torch.Tensor:
-    """Test application to a rectangular PIL image."""
-    transform = ColOutTransform(p_row, p_col)
-    new_image = transform(fake_image)
-    verify_shape_image(fake_image, new_image, p_row, p_col)  # type: ignore
+class TestColOutTransform:
 
+    def test_single_image_drop_size(self, fake_image: Image.Image, p_row: float, p_col: float):
+        """Test application to single PIL image."""
+        transform = ColOutTransform(p_row, p_col)
+        new_image = transform(fake_image)
+        verify_shape_image(fake_image, new_image, p_row, p_col)
 
-def test_single_image_tensor_drop_size(fake_image_tensor: torch.Tensor, p_row: float, p_col: float):
-    """Test application to a single torch image tensor."""
-    transform = ColOutTransform(p_row, p_col)
-    new_image = transform(fake_image_tensor)
-    verify_shape_tensor(fake_image_tensor, new_image, p_row, p_col)  # type: ignore
+    @pytest.mark.parametrize("W", [48])
+    def test_rectangular_image(self, fake_image: Image.Image, p_row: float, p_col: float):
+        """Test application to a rectangular PIL image."""
+        transform = ColOutTransform(p_row, p_col)
+        new_image = transform(fake_image)
+        verify_shape_image(fake_image, new_image, p_row, p_col)  # type: ignore
 
+    def test_single_image_tensor_drop_size(self, fake_image_tensor: torch.Tensor, p_row: float, p_col: float):
+        """Test application to a single torch image tensor."""
+        transform = ColOutTransform(p_row, p_col)
+        new_image = transform(fake_image_tensor)
+        verify_shape_tensor(fake_image_tensor, new_image, p_row, p_col)  # type: ignore
 
-def test_reproducibility_image(fake_image_tensor: torch.Tensor, p_row: float, p_col: float):
-    """Test that transform is reproducible given the same seed."""
-    transform_1 = ColOutTransform(p_row, p_col)
-    transform_2 = ColOutTransform(p_row, p_col)
+    def test_reproducibility_image(self, fake_image_tensor: torch.Tensor, p_row: float, p_col: float):
+        """Test that transform is reproducible given the same seed."""
+        transform_1 = ColOutTransform(p_row, p_col)
+        transform_2 = ColOutTransform(p_row, p_col)
 
-    torch.manual_seed(42)
-    new_image_1 = transform_1(fake_image_tensor)
-    torch.manual_seed(42)
-    new_image_2 = transform_2(fake_image_tensor)
+        torch.manual_seed(42)
+        new_image_1 = transform_1(fake_image_tensor)
+        torch.manual_seed(42)
+        new_image_2 = transform_2(fake_image_tensor)
 
-    assert torch.allclose(new_image_1, new_image_2)
+        assert torch.allclose(new_image_1, new_image_2)
 
 
-def test_reproducibility_batch(fake_image_batch: torch.Tensor, p_row: float, p_col: float):
-    """Test that batch augmentation is reproducible given the same seed."""
-    transform_1 = functools.partial(colout_batch, p_row=p_row, p_col=p_col)
-    transform_2 = functools.partial(colout_batch, p_row=p_row, p_col=p_col)
+class TestColOutFunctional:
```

Comment:
tests of the functional API and those of the algorithm class are now separately grouped for better readability.